lib/cron/cache-pools.ts

```diff
@@ -1,11 +1,17 @@
+import { Protocol } from '@uniswap/router-sdk'
+import { ChainId, V2SubgraphProvider, V3SubgraphProvider } from '@uniswap/smart-order-router'
 import { EventBridgeEvent, ScheduledHandler } from 'aws-lambda'
 import { S3 } from 'aws-sdk'
 import { default as bunyan, default as Logger } from 'bunyan'
-import _ from 'lodash'
 import { S3_POOL_CACHE_KEY } from '../util/pool-cache-key'
-import { chainProtocols } from './cache-config'
 
 const handler: ScheduledHandler = async (event: EventBridgeEvent<string, void>) => {
+  const chainId: ChainId = parseInt(process.env.chainId!)
+  const protocol = process.env.protocol! as Protocol
+  const timeout = parseInt(process.env.timeout!)
+  // Don't retry for V2 as it will timeout and throw 500
+  const provider =
+    protocol === Protocol.V3 ? new V3SubgraphProvider(chainId, 3, timeout) : new V2SubgraphProvider(chainId, 1, timeout)
```

Comment:
Why do we reinitialize the subgraph provider here instead of just using the providers created in cache-config? It would be simpler to just pass in chainId and protocol and do a lookup from cache-config?